e2e/test/scenarios/actions/actions-in-object-detail-view.cy.spec.js

```diff
@@ -0,0 +1,387 @@
+import moment from "moment-timezone";
+
+import { WRITABLE_DB_ID } from "e2e/support/cypress_data";
+import {
+  modal,
+  popover,
+  resetTestTable,
+  restore,
+  resyncDatabase,
+  undoToast,
+  visitDashboard,
+  visitModel,
+} from "e2e/support/helpers";
+
+const PG_DB_ID = 2;
+const PG_SCOREBOARD_TABLE_ID = 9;
+const WRITABLE_TEST_TABLE = "scoreboard_actions";
+const FIRST_SCORE_ROW_ID = 11;
+const SECOND_SCORE_ROW_ID = 12;
+const UPDATED_SCORE = 987654321;
+const UPDATED_SCORE_FORMATTED = "987,654,321";
+
+const SCORES_MODEL = {
+  name: "Scores model",
+  dataset: true,
+  display: "table",
+  database: PG_DB_ID,
+  query: {
+    "source-table": PG_SCOREBOARD_TABLE_ID,
+  },
+};
+
+const DASHBOARD = {
+  name: "Test dashboard",
+  database: PG_DB_ID,
+};
+
+describe("scenarios > actions > actions-in-object-detail-view", () => {
+  beforeEach(() => {
+    cy.intercept("POST", "/api/action").as("createBasicActions");
+    cy.intercept("GET", "/api/action?model-id=*").as("getModelActions");
+    cy.intercept("GET", "/api/action/*/execute?parameters=*").as(
+      "prefetchValues",
+    );
+
+    resetTestTable({ type: "postgres", table: WRITABLE_TEST_TABLE });
+    restore("postgres-writable");
+  });
+
+  describe("in dashboard", () => {
+    beforeEach(() => {
+      asAdmin(() => {
+        resyncDatabase({
+          dbId: WRITABLE_DB_ID,
+          tableName: WRITABLE_TEST_TABLE,
+        });
+
+        cy.createQuestion(SCORES_MODEL, { wrapId: true, idAlias: "modelId" });
+
+        cy.get("@modelId").then(modelId => {
+          createBasicModelActions(modelId);
+
+          cy.createQuestionAndDashboard({
+            questionDetails: {
+              name: "Score detail",
+              display: "object",
+              database: PG_DB_ID,
+              query: {
+                "source-table": `card__${modelId}`,
+              },
+            },
+            dashboardDetails: DASHBOARD,
+          }).then(({ body: { card_id, dashboard_id } }) => {
+            cy.wrap(card_id).as("modelId");
+            cy.wrap(dashboard_id).as("dashboardId");
+          });
+        });
+      });
+    });
+
+    it("does not show model actions in model visualization on a dashboard", () => {
+      asAdmin(() => {
+        cy.get("@dashboardId").then(dashboardId => {
+          visitDashboard(dashboardId);
+        });
+
+        cy.findByTestId("dashcard").within(() => {
+          assertActionsDropdownNotExists();
+        });
+      });
+    });
+  });
+
+  describe("in modal", () => {
+    beforeEach(() => {
+      asAdmin(() => {
```

Comment:
I found these `asAdmin`/`asNormalUser` function helpful in organizing/reading the code in this test suite thanks to extra indentation for a user type dimension.

Please let me know if this needs an RFC so I'll revert it (but trust me, it will hurt the readability of this test suite) - easy change. I can also move them to live with other helpers, since they're useful everywhere.